### ISSUE TITLE
The plugin does not detect proto file was changed when running build for a second time

### ISSUE THREAD

**Issue by dapengzhang0** [google/protobuf-gradle-plugin#331]
Use the gradle plugin to generate code, then change the proto file, then build again.
As reproduced in https://sponge.corp.google.com/target?id=ff244382-5595-48fd-b556-4b2694aaf5d4&target=grpc%2Fjava%2Fmaster%2Fpresubmit%2Fandroid&searchFor=

**Comment by ejona86** [comment]
In general, Gradle notices when files have been deleted and recompiles. There's basically no change that is "the application's responsibility".

**Comment by voidzcy** [comment]
Filed #332, which should be done first before we can make changes to always wipe out generated file directories.

**Comment by devinrsmith** [comment]
I've ran across this just now (after a large update to our proto files). IMO, this should be the responsibility of the plugin. 

Hopefully anybody else in this situation can use a workaround like this for now:

```
protobuf {
    protoc { artifact = "com.google.protobuf:protoc:${protocVersion}" }
    generateProtoTasks {
      all().each {
        generateProtoTask ->
          // The generateProto task does not seem to properly clean its previously generated outputs.
          // See https://github.com/google/protobuf-gradle-plugin/issues/332
          // See https://github.com/google/protobuf-gradle-plugin/issues/331
          generateProtoTask.doFirst {
            delete generateProtoTask.outputs
          }
      }
    }
  }
```

**Comment by joschi** [comment]
I've been running into the same problem and worked around it as described in https://github.com/google/protobuf-gradle-plugin/issues/331#issuecomment-543333726.

Any chance of getting this PR merged?

**Comment by clayburn** [comment]
Is this still an issue? I couldn't reproduce, although maybe I'm missing a step here.

**Comment by ejona86** [comment]
Duplicate of #33
